Favicon as status

Charon77 opened the pull request:
In response to:

https://github.com/syncthing/syncthing/issues/1018

https://www.bountysource.com/issues/6486139-use-favicon-as-an-indicator-of-syncthing-status

calmh commented:
Thanks! Can you explain the change to .gitattributes?

AudriusButkevicius commented:
Also, the indent should be spaces not tabs.

calmh commented on gui/syncthing/core/eventService.js:
```diff
@@ -18,14 +18,45 @@ angular.module('syncthing.core')
                     $rootScope.$broadcast(event.type, event);
+					
+					if(event.type=='FolderErrors')
+					{
```
Brace style on these, should follow the rest of the code base.

Charon77 commented:
Hi, @calmh 
The only thing I changed in .gitattributes is, I made it so that it doesn't convert any line endings. SyncThing didn't compile under Windows prior that, because of this:

https://forum.syncthing.net/t/having-problems-building-syncthing-on-windows-10/6188

I think it is not that important and thus can be ignored.

@AudriusButkevicius  Ah, so sorry about that.

calmh commented:
I doubt that gitattributes change is correct, and regardless it belongs in a separate PR if that's the case.

Charon77 commented:
@calmh you are right. I have reverted the .gitattributes. The 'if spacing' was due to tabs instead of space. I have changed that as well. 

Could you please elaborate on:
"Can we just add a colored bubble to the normal logo perhaps"
Is it like, a static tiny green bubble, or something different?

Thank you.

calmh commented:
I'm not sure, I guess I'd need to try it to see what makes sense at the small size these are displayed. I guess I don't really like a red Syncthing logo because it's the logo and it's supposed to be blue; but it can also be monochrome, so how about for example a grey logo with a red/green/blue indicator of some sort, i.e. a "blob"?

<img width="87" alt="screen shot 2015-12-26 at 16 40 10" src="https://cloud.githubusercontent.com/assets/125426/12006995/6d826230-abef-11e5-9991-96b59fbede06.png">

Maybe that doesn't work at this size and then maybe coloring the logo is the right way to go.

calmh commented on gui/syncthing/core/eventService.js:
```diff
@@ -18,14 +18,42 @@ angular.module('syncthing.core')
                     }
                     $rootScope.$broadcast(event.type, event);
+                    
+                    if(event.type=='FolderErrors') {
```
Also spacing, as in other if/for/etc things in this file.

calmh commented on gui/index.html:
```diff
@@ -14,7 +14,7 @@
   <meta name="description" content="">
   <meta name="author" content="">
-  <link rel="shortcut icon" href="assets/img/favicon.png">
+  <link rel="shortcut icon" href="assets/img/favicon_idle.png">
```
Maybe give this one an `id` to avoid all the `link[rel='shortcut icon']`?

Charon77 replied:
Alright.

Charon77 commented:
@calmh  I'm going to make some more changes to the JS. Meanwhile, do check the mockup icons I made.

Cheers :)

calmh commented:
@Charon77 Those look better to me, and you can probably get away with having a larger blob to make the state more obvious. I guess it depends on how it looks in the actual browser.

AudriusButkevicius commented on gui/syncthing/core/eventService.js:
```diff
@@ -18,14 +18,43 @@ angular.module('syncthing.core')
+            
             if (lastID > 0) {   // not emit events from first response
+                var green_counter=0;
+                var red_counter=0;
```
redCounter rather than red_counter, to make it consistent.
Or just red and green.

AudriusButkevicius commented on gui/syncthing/core/eventService.js:
```diff
@@ -18,14 +18,43 @@ angular.module('syncthing.core')
             $rootScope.$broadcast(self.ONLINE);
-
+            //GREEN
+            $("link[rel='shortcut icon']").attr("href","assets/img/favicon_idle.png")
```
I suggest you do 

var statusIcon = $(...selector...);

and reuse that, as otherwise you are traversing the DOM every time you want to change something.

AudriusButkevicius replied:
Also, this specific modification will always be overriden by the else BLUE branch.

AudriusButkevicius replied:
Also missing `;` on a lot of lines.

AudriusButkevicius commented on gui/syncthing/core/eventService.js:
```diff
@@ -18,14 +18,43 @@ angular.module('syncthing.core')
                     }
                     $rootScope.$broadcast(event.type, event);
+                    
+                    if (event.type=='FolderErrors') {
```
You should use the enum values, or define enum values if they are not defined. Same like self.OFFLINE, etc.

Charon77 commented:
@AudriusButkevicius I've fixed the writing styles. I hope I didn't miss any.

@calmh Which icon should be used? Or is any icon need changing?

fti7 commented:
I would prefer the original Logo with different Colors instead of the bubbles.
On Firefox the Favicons are already very small...

I would also like to suggest to define a "official" Status Icon behavior... so it could be implemented the same on the 3rd Party (GUI) Tool as well.

Synctrazor's Windows Tray Icon uses some nice behaviors (But didnt cover all States yet)

Current:
Grey: Syncthing not running
Blue Icon: Idle
Blue Rotating Icon: Syncing
-> But the Rotation Movement just had 3 Positions
This is more calmed down, not like the green Icon in your PR

See
https://github.com/canton7/SyncTrayzor/tree/master/src/SyncTrayzor/Icons

### Icon behavior Proposal
Grey: Syncthing not running (Web GUI Connection Error? Or should we treat this as an Error?)
Blue: Idle
Blue Rotating (3 Positions): Syncing
Red: Error
Green (Static or with an Pulsating Effect): User Intervention needed (New Folder, Device etc)

BR,
Frank

Charon77 commented:
Hi, @fti7, thank you for your input.
That does look nice, and I do admit that the rotation looks cool. However, as @calmh  has mentioned, changing the logo color to red and green might need some further consideration as it is the logo we'll be messing with. I await further decision, of which I would act upon on.

AudriusButkevicius commented:
IMHO, gray and blue is fine, for other states as suggested we could motion.
Rotation for syncing, bouncing sideways for error, shrinking and growing in size for requiring user intervention.

calmh commented:
Well, I'm not a graphic designer, but here's as I see it as far as concerns the logo:

 - Status overlays on top of the logo: sure, why not. (This may not be suitable for the favicon for size reasons)
 - Monochrome:ing the logo to indicate stopped / not active etc: fine
 - Rotation: Not something I'll merge; it has a natural orientation and frankly this looks tacky to me. Bouncing / resizing may be ok.

Again, the favicon is a super restricted space to work with, I get that; otherwise I much prefer things like these (from a google of similar discussions): https://bugzilla.mozilla.org/attachment.cgi?id=429115, https://bugzilla.mozilla.org/attachment.cgi?id=434506, https://bugzilla.mozilla.org/attachment.cgi?id=438739 - that is, overlays that indicate something is happening.

Charon77 commented:
@calmh 
Do you have any comments regarding the new icons? They are all based on blue / grayscale.

andersonvom commented:
Gmail has a lab which allows you to add the number of new messages to the favicon. Although small, you can read the number fairly well. I think if we wanted to use a blob with colors/letters it should work fine as well.

<img width="27" alt="screen shot 2015-12-26 at 12 57 53" src="https://cloud.githubusercontent.com/assets/69922/12007426/062cffb8-abd2-11e5-90be-95d9eae37920.png">

AudriusButkevicius commented:
So I think the overlay icons are much better, minus the 3D rotation and the yellowish colour.

Charon77 commented:
I've added more icons. Feel free to drop comments.

Charon77 commented:
@AudriusButkevicius  @calmh  
Hi, I wanted to know if there's any possible improvement for this branch.
Thank you.

uok commented:
Thanks for working on this old idea of mine!
My suggestions: 
- the base icon should always the same (blue) Syncthing icon to improve usability (especially if you have more tabs pinned it is easier to find for your eyes)
- the overlay shows what's happening (error, sync, user interaction required)
- to keep things calm only sync should have an animated overlay
- overlay should be the same size on all different states (currently overlay rotating arrows is a little bigger than the overlay with exclamation mark)

Charon77 commented:
@uok Hi, and you're welcome!

On the case of changing the icon color, we might need to convince either @AudriusButkevicius  or @calmh to let it through. Right now, the only option given is to use either the standard blue, or the inactive gray.

Yes, I have used some Chrome-derived browser as well, which displays the same problem, i.e. only displaying first frame of GIF. Maybe they will change that some day in the future; who knows?

uok commented:
ok, what about this compromise to take care of small favicons / non-animation support:
base icon always blue, overlay is a big dot/circle with different colors (no exclamation sign or arrows)
- idle: no overlay
- error: red indicator 
- sync: green indicator (maybe blinking)

I'm also thinking about moving the indicator to the top right corner. 
Another idea is to color the corner making the indicator more a triangle shape

Charon77 commented:
@uok  Something like that? ![](https://github.com/Charon77/syncthing/blob/favicon-status/gui/assets/img/favicon_syncing4.gif?raw=true)

fti7 commented:
+1 for the first icon (Blue ST Logo with small colored dot)

For syncing: Please no blinking icon. The Syncing state should not trigger/annoy to look into the Syncthing UI.

What i would like to have is an State where User Intervention is needed (New Folder, Device etc)
This could/should be a blinking one


@canton7 What are you thinking about this? (For SyncTrayzor)

canton7 commented:
I'm keeping an eye on this thread :wink:. If the conclusions look good, I'll steal them for SyncTrayzor. 

I use a rotating version of the blue syncthing icon to mean "synchronising", a grey version to mean "syncthing not started", and a steady blue icon for everything else. This works pretty well. 

I was considering blue with a small red exclamation mark to mean "user attention required".

uok commented:
since some users have set 60s scan intervall I agree that pulsating icons would be more annoying than helpful. My suggestion: keep it simple - if you pin the Syncthing tab you should easily see the current status: no activity -- syncing -- interaction required -- error
![indicate circle](https://cloud.githubusercontent.com/assets/9367272/12021805/15fc0826-ad8a-11e5-89c4-a4fefb8baed7.png) or ![indicate square](https://cloud.githubusercontent.com/assets/9367272/12021809/191088de-ad8a-11e5-8497-1d5f100f9f0e.png)

uok commented on gui/syncthing/core/eventService.js:
```diff
@@ -18,14 +28,42 @@ angular.module('syncthing.core')
             if (lastID > 0) {   // not emit events from first response
+                var greenCounter = 0;
+                var redCounter = 0;
+                
```
what about "var iconstate = 'idle'" as default

Charon77 replied:
I don't really get what is it for

uok replied:
instead of counting each color in a separate variable you should use a single variable that holds the icon state, e.g. 'error'

uok commented on gui/syncthing/core/eventService.js:
```diff
@@ -18,14 +28,42 @@ angular.module('syncthing.core')
+                    if (event.type == self.STATE_CHANGED && event.data.to == 'syncing') {
+                        greenCounter++;
+                    }
+                    
```
maybe this could be simplified with a default state (see above) and then change the state accordingly, e.g.

    if (event.type == self.FOLDER_ERRORS) {
      iconstate = 'error';
    }
    elseif (other conditions) {
      ...
    }
    changeFavIcon(iconstate);

this would make the error state superseed the rest, then the next, etc. etc. - no further counting needed

uok commented on gui/syncthing/core/eventService.js:
```diff
@@ -6,6 +6,16 @@ angular.module('syncthing.core')
+        
+        function changeFavIcon (icon) {
+            shortcutIcon.attr("href","/assets/img/" + icon);
+        }
```
maybe simply to

    shortcutIcon.attr("href","/assets/img/favicon_" + state + ".png");

since all icons have the similar file names

Charon77 replied:
If we were going to drop gifs, or drop pngs, I guess that would be better.

Charon77 commented:
I have done some more refactoring.
Also, do check the latest addition to the logo files, and feel free to comment..

zertrin commented:
Just 50 cents here: if planning to use color to distinguish different statuses, please think about colorblind people (with all degrees of colorblindness that exist, this is a non negligible part of the population :wink:)

Also I'd rather be careful with red/yellow/green, except if their luminance is different enough (i.e. dark red / bright yellow / not-too-bright-or-dark green)

AudriusButkevicius commented:
I guess I prefer the version with circle and symbols in the circle, the circle potentially having a color, and potentially no animations to make it less annoying.

This is becoming a bikeshedding thread, so @calmh needs to put his foot down and decide.

calmh commented:
Okay, here comes the foot. I've looked around for prior art, and the most clear and tasteful I found in my immediate vicinity is the Dropbox menu bar item. In the normal state (connected, in sync), it's just a black logo. It being monochrome is the Mac menubar style, so we could/would be blue in the same state.

<img width="38" alt="normal" src="https://cloud.githubusercontent.com/assets/125426/12062677/258817e4-afa1-11e5-8c30-79069b8747d3.png">

(Ignore that it's all kind of brown; that translucency to my desktop background...)

When manually paused, it dims the logo and displays a pause symbol.

<img width="66" alt="paused-l" src="https://cloud.githubusercontent.com/assets/125426/12062676/1d0cd69a-afa1-11e5-84f9-ad488b883e1e.png">

In my mind this could equal our "stopped" state; potentially with a "stop square" instead of the pause bars, but it's roughly the same. The dimming of the logo makes it at a glance different and highlights both that it's not doing it's thing as usual and that there is an extra symbol.

When syncing, it's non-dimmed and shows a syncing symbol.

<img width="34" alt="screen shot 2015-12-31 at 09 21 54" src="https://cloud.githubusercontent.com/assets/125426/12062666/de51c05a-afa0-11e5-8616-786d16d49db8.png">

This indicates both normal operation (not dimmed) and something happening (there's a symbol).

Dropbox also has an additional state when there is no network connectivity:

<img width="52" alt="disabled-l" src="https://cloud.githubusercontent.com/assets/125426/12062681/55e2bcdc-afa1-11e5-8bd7-83dba0d8bb40.png">

I'm not sure this maps against something on our side. Possibly "all devices are disconnected" or something.

So, I'd very much like something similar. That is:

 - Logo in normal color plus dimmed (dimmed could be greyscale with reduced contrast, for example)
 - Small, simple symbol that is not dimmed and stands out a little against the background logo.
 - No colors necessary, so no color blindness issues.
 - No animations necessary (although the syncing symbol could well be animated, no strong opinion here.)

This is almost as some of the variants above, except coloring, separation of the symbol against the underlying logo, things like that. Makes sense? I may give this an attempt, but it would probably be altogether better if a pro like our resident graphical magician @gillisig took a look. ;)

uok commented:
@zertrin good point

I toyed around with different ideas - how many different states do we even need? I think besides "idle" only "error" and "interaction" are necessary, because only then the user needs to do something / be notified. Otherwise Syncthing does its job, no need for a "syncing" state.

We could even reduce it to "idle" and "notification" which should be enough for everyday use. So whenevery something is different from "idle / syncing" the user is informed, e.g. incoming request, error, out of space, etc.

regular ![text4204-9-87](https://cloud.githubusercontent.com/assets/9367272/12063392/4311f1ce-afae-11e5-904d-11a8ae19f6c9.png) or black/white ![text4204-9-8](https://cloud.githubusercontent.com/assets/9367272/12063390/4311442c-afae-11e5-90ad-2111548c60e8.png) or lighten ![text4204-9](https://cloud.githubusercontent.com/assets/9367272/12063391/4311744c-afae-11e5-8ed2-32db4c0d9e99.png)

gillisig commented:
I agree with @calmh, the icon shouldn't draw too much attention to it, so rotating, pulsing or something like that is out. 

It just so happens that I actually installed an app moments ago, which added a rotating icon to my task-bar when it was working. It kept drawing my eye too it again and again. Movement tends to draw your eyes to it, so this ends up being too distracting for most users who just want to get some work done. Let's not annoy them too much. 

I understand perfectly well that people often really want make it spin, pulse or animate in some way. Because its kind of cool, for a brief moment. Like the extreme animations you see in the Windows phones. But then it very quickly it becomes annoying. 

Changing color of the whole icon might also not serve the purpose we hope it would, since people most often associate an app with a certain shape and color. So when my blue Syncthing icon is red all of the sudden, I might not find it. I have seen this happen a lot when I do user testing, users keep searching for something expecting it to look exactly the same way they are used to, but end up not finding it because it changed a little. This is often a big problem during redesigns, improvements often take some getting used to, not because they are better or worse but simply because they are different from what the user is used to. So making the whole icon that you use to find Syncthing change colors frequently would likely just frustrate the user. Just imagine for a second if the github icon was red all of the sudden, would you be able to find the tab quickly? Or would you have to spend an extra few seconds to realise what happened. 

Anyway I hope this helps explain why what @calmh suggested might be the best solution. 

I can take a stab at making one myself but I am not quite sure when I will have the time. I will try to squeeze it in soon though.

In the meantime anyone else is of course free to do so themselves. We should probably make two versions though, colored and grayscale because mac/linux usually use grayscale icons but windows/web uses colored.

canton7 commented:
FWIW I've had a rotating icon to mean "syncing" in SyncTrayzor from day 1, and not a single user has ever complained or even mentioned it...

AudriusButkevicius commented:
To me syncthing icon rotating looks like and old wooden carriage wheel from a looney tunes movie.

uok commented:
@canton7 I think this is fine for the system tray
@AudriusButkevicius :laughing:

indrora commented:
Figured I'd throw some brain at it. Here's two options, plus (because I was feeling flat) a flattened version of the logo.

Error:
![error_a_16](https://cloud.githubusercontent.com/assets/470584/12739493/7b4c540c-c927-11e5-9825-b4dba3c46bff.png)![error_a_24](https://cloud.githubusercontent.com/assets/470584/12739497/7b5062b8-c927-11e5-9307-9778cecb2d82.png)![error_a_32](https://cloud.githubusercontent.com/assets/470584/12739496/7b505958-c927-11e5-912b-dd062016b282.png)

![error_b_16](https://cloud.githubusercontent.com/assets/470584/12739498/7b511848-c927-11e5-9a1f-a78c94aef40e.png)![error_b_24](https://cloud.githubusercontent.com/assets/470584/12739495/7b50325c-c927-11e5-9f22-656825e35334.png)![error_b_32](https://cloud.githubusercontent.com/assets/470584/12739494/7b4f1868-c927-11e5-9adf-4656e121d551.png)

Prompt:
![prompt_a_16](https://cloud.githubusercontent.com/assets/470584/12739536/c67baefa-c927-11e5-8e5f-1780d86c69cf.png)![prompt_a_24](https://cloud.githubusercontent.com/assets/470584/12739541/ca1a80e0-c927-11e5-85d7-a604be1d587a.png)![prompt_a_32](https://cloud.githubusercontent.com/assets/470584/12739542/cf9e1914-c927-11e5-8822-4f53338b5fb3.png)


![prompt_b_16](https://cloud.githubusercontent.com/assets/470584/12739551/dce8b03e-c927-11e5-895f-3060f543a852.png)![prompt_b_24](https://cloud.githubusercontent.com/assets/470584/12739553/dd2b6f50-c927-11e5-9db1-430b9fac3060.png)![prompt_b_32](https://cloud.githubusercontent.com/assets/470584/12739552/dd296b38-c927-11e5-9d93-b20596c09388.png)

Synchronizing:

![sync_a_16](https://cloud.githubusercontent.com/assets/470584/12739559/f053b48e-c927-11e5-81c7-708d11d5e068.png)![sync_a_24](https://cloud.githubusercontent.com/assets/470584/12739576/16e70b0a-c928-11e5-93d4-0d58c5c6b232.png)![sync_a_32](https://cloud.githubusercontent.com/assets/470584/12739560/f053c7e4-c927-11e5-9736-ba7c5eaa1598.png)

![sync_b_16](https://cloud.githubusercontent.com/assets/470584/12739557/f051ac7a-c927-11e5-99d7-8898d239381a.png)![sync_b_24](https://cloud.githubusercontent.com/assets/470584/12739561/f05b9c12-c927-11e5-9258-bae264315f4f.png)![sync_b_32](https://cloud.githubusercontent.com/assets/470584/12739558/f053a624-c927-11e5-8afd-323871f2b81d.png)

And as a bonus, a flat icon (because I couldn't find an SVG version)

![icon_256](https://cloud.githubusercontent.com/assets/470584/12739587/2a50c3f2-c928-11e5-9c35-f6126d64f2dd.png)![icon_32](https://cloud.githubusercontent.com/assets/470584/12739589/2a52506e-c928-11e5-8ca2-29cfd6a96c75.png)
![icon_48](https://cloud.githubusercontent.com/assets/470584/12739595/344a739e-c928-11e5-8859-394a1d037c76.png)![icon_96](https://cloud.githubusercontent.com/assets/470584/12739588/2a51c946-c928-11e5-9d7c-541cc9c1def0.png)

calmh commented:
I still agree with the idea of this, but none of the proposed alternatives so far fit the simplicity and clarity of the dropbox example, in my opinion.

vontrapp commented:
I think it would be cool to have the nodes in the graph of the logo indicate status. Perhaps the nodes blinking alternately while syncing, red lines between nodes on error, and [blinking] yellow middle node for user action. Maybe also slightly enlarging the nodes when colored/relaying status.

lkwg82 commented:
I'd like to close this as stalled.

canton7 commented:
There's good work and discussion in here - let's not let it be drowned in the heap of closed issues

lkwg82 commented:
Of course, this means the label. But this is stalled for more than 2 months. How can we get further?

canton7 commented:
That's not a metric for whether an issue should be closed. There are loads of open issues under the 'Unplanned' milestone that are waiting for someone to get around to them: that doesn't mean that any of them should be closed.

Closed issues are either things which have been resolved, duplicates, or suggestions which have no merit. Reasonable features or unresolved bugs which just need someone's time don't fit into that category.

AudriusButkevicius commented:
This is a PR that has stalled and has been closed, I think the issue is still open.

norgeous commented:
How about this one?

![favicon64x64](https://cloud.githubusercontent.com/assets/4681349/15635710/4ae9b2fa-25de-11e6-8fc7-452b11e67467.png)
64x64

![favicon16x16-attempt2](https://cloud.githubusercontent.com/assets/4681349/15635704/25e3da08-25de-11e6-848d-85c990b3841d.png)
16x16 (Lanczos3)

I can make svg(s) available or a PR if this is liked.

AudriusButkevicius commented:
Looks good to me.

norgeous commented:
![cleaned16](https://cloud.githubusercontent.com/assets/4681349/15639554/ed70ac04-2624-11e6-8e3d-0a443a9c2580.png) Default / Idle
![progress](https://cloud.githubusercontent.com/assets/4681349/15639393/e7f898ce-2622-11e6-9f53-e5b66aab4b6e.png) Syncing
![mono](https://cloud.githubusercontent.com/assets/4681349/15639398/f28045a8-2622-11e6-8cc8-3ae4fd54f32a.png) ![pause-mono](https://cloud.githubusercontent.com/assets/4681349/15639401/f5cd2546-2622-11e6-98ab-3e9cf88ff2e0.png) Paused with monotone logo
![transparent](https://cloud.githubusercontent.com/assets/4681349/15639402/f7bcf7fa-2622-11e6-87d8-f2fad5be61d3.png) ![pause-transparent](https://cloud.githubusercontent.com/assets/4681349/15639403/f9bed47e-2622-11e6-84ae-ba453a12c2a7.png) Paused with 50% transparent logo
![error-yellow](https://cloud.githubusercontent.com/assets/4681349/15639406/fd52ac50-2622-11e6-92c4-861ae4bfb297.png) Notify
![error-red](https://cloud.githubusercontent.com/assets/4681349/15639407/00552f72-2623-11e6-80bf-89c06ff9ff89.png) Error

Some are [animated png](http://caniuse.com/#feat=apng) so in chrome and opera you only get the first frame. I have experimented with using gif for the favicon, but it lacks the necessary alpha channel support, and the transparent edges look strange.

For anyone interested, I export a 1600x1600 png from inkscape then lanczos3 it back down to 16x16 in gimp for best effect. Animation is done with [APNG Assembler](http://littlesvr.ca/apng/assembler/assembler.php).

All done in svg, so can be good quality, I have tried to make my additions line up with an invisible 16x16 grid, so that when scaled down to 16x16 (as above) things remain clear - as pixels are aligned.

![untitled](https://cloud.githubusercontent.com/assets/4681349/15639815/60e4098a-2628-11e6-911f-cdd4a7b2bd32.png)

![error-flash](https://cloud.githubusercontent.com/assets/4681349/15639408/0244c40a-2623-11e6-92ab-635edd2a92b6.png) for those who said it wasn't eye catching enough

norgeous commented:
![screenshot at 2016-05-30 06 20 14](https://cloud.githubusercontent.com/assets/4681349/15640392/5bf5dc62-262f-11e6-822c-3b5dc19e23fa.png) they work on dark too...

not sure about the black pause button tho.

uok commented:
Thanks for working on this! Can you please post a version on light regular tabs (e.g. IE or Firefox)
My vote is for a pause icon just gray (without bars)
I'm also not sure if we need extra warning and error version, why not just keep error version as a general notifiy icon (= user has to do something)? so "idle", "sync", "pause" and "notify"

norgeous commented:
Can't post a version of normal Firefox now, just shutting down... although you can preview each in Firefox, by right clicking > Open Link in New Tab, then Pin the tab as normal. Right click the individual pngs in the earlier post, not the ones in the screenshot with the dark bg.

calmh commented:
I like those versions. I prefer non animated, but that's somethin I'll fight to the death over.

norgeous commented:
> I like those versions.

I will prepare a PR

AudriusButkevicius commented:
Superseded by #3217
